src/Microsoft.OData.Core/UriParser/TypePromotionUtils.cs

```diff
@@ -261,6 +261,13 @@ internal static void GetTypeFacets(IEdmTypeReference type, out int? precision, o
                     return true;
                 }
 
+                // Comparing an enum with a string is valid
+                if (left.IsEnum() && right.IsString())
+                {
+                    right = left;
+                    return true;
+                }

```

Comment:
Any reason not to support the same comparison with the string on the left and enum on the right?

Reply:
That looks like a valid case. Added logic for conversion when string is on left